src/Netscript/APIWrapper.ts

```diff
@@ -129,7 +148,7 @@ export function wrapAPI(
     if (typeof value === "function") {
       wrapFunction(helpers, wrappedAPI, workerScript, value, ...tree, key);
     } else if (Array.isArray(value)) {
-      setNestedProperty(wrappedAPI, value, key);
+      setNestedProperty(wrappedAPI, value.slice(), key);
```

Comment:
whats your reasoning for doing a slice here?

Reply:
I was trying to address a separate bug where scripts do not end gracefully if the player mutates the args array. This didn't end up being a solution but I think it still makes sense not to give players access directly to an actual array the game uses internally, so the player just gets a copy of it.

Reply:
ok, honestly at somepoint i want to put a sanitizer in this wrapper that does a deepCopy on any return values to prevent that completely, but this seems fine for now